src/main/java/mate/academy/service/impl/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,45 @@
+package mate.academy.service.impl;
+
+import mate.academy.dao.ShoppingCartDao;
+import mate.academy.dao.TicketDao;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.MovieSession;
+import mate.academy.model.ShoppingCart;
+import mate.academy.model.Ticket;
+import mate.academy.model.User;
+import mate.academy.service.ShoppingCartService;
+
+@Service
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    @Inject
+    private ShoppingCartDao shoppingCartDao;
+    @Inject
+    private TicketDao ticketDao;
+
+    @Override
+    public void addSession(MovieSession movieSession, User user) {
+        Ticket ticket = new Ticket(movieSession, user);
+        ticketDao.add(ticket);
+        ShoppingCart shoppingCartByUser = getByUser(user);
+        shoppingCartByUser.getTickets().add(ticket);
+        shoppingCartDao.update(shoppingCartByUser);
+    }
+
+    @Override
+    public ShoppingCart getByUser(User user) {
+        return shoppingCartDao.getByUser(user).orElseThrow(() -> new RuntimeException("Can't get "
```

Comment:
```suggestion
        return shoppingCartDao.getByUser(user).orElseThrow(() -> new NoSuchElementException("Can't get "
```
On DAO layer the type of exception is correct